scripts/login_page.js

```diff
@@ -0,0 +1,194 @@
+    } else if (pass_check === false) {
+        check_pass();
+
+    } else if (confirm_pass_check === false) {
+        check_confirm_pass();
+    } else final_sign_up_check();
+}
+
+function check_name() {
+    user_nickname = get_value("id", "name");
+    name_validation();
+}
+
+function name_validation() {
+    name_check = false;
+    show_element("id", "name_notifications_field");
+
+    if (user_nickname === 'Lex') {
+        send_notification("id", "name_notifications_field", "Name accepted");
+        set_class_name("id", "name_notifications_field", "success_message");
+        wait(hide_message, "name_notifications_field", 2000);
+        show_element("class", "pass_group");
+        return name_check = true;
+
+    } else if (user_nickname === "") {
+        notifications_field.innerHTML = 'At first enter your name';
+        notifications_field.className = 'error_message';
+    } else {
+        notifications_field.innerHTML = 'User not found';
+        notifications_field.className = 'error_message';
+    }
+}
+
+function check_pass() {
+    password = get_value("id", "pass");
+    notifications_field = get_element("id", "pass_notifications_field");
+    pass_check = false;
+    pass_check = password_validation(password);
+}
+
+function ask_confirm_pass() {
+    show_element("class", "confirm_pass_group");
+}
+
+function check_confirm_pass() {
+    confirm_pass_check = false;
+    notifications_field = get_element("id", "confirm_pass_notifications_field");
+    password = get_value("id", "pass");
+    confirm_password = get_value("id", "confirm_pass");
+    password_validation(confirm_password);
+    compare_passwords();
+
+    if (
+        password_validation(confirm_password) === true
+        && compare_passwords() === true
+    ) {
+        confirm_pass_check = true;
+        next_step();
+    }
+}
+
+function compare_passwords() {
+    notifications_field = get_element("id", "confirm_pass_notifications_field");
+    if (password === confirm_password) {
+        return true;
+    } else {
+        notifications_field.innerHTML = "Confirm password does not match the Password";
+    }
+}
+
+function final_sign_up_check() {
+    user_nickname = get_value("id", "name");
+    password = get_value("id", "pass");
+    confirm_password = get_value("id", "confirm_pass");
+    if (
+        name_validation() === true
+        && password_validation(password) === true
+        && password_validation(confirm_password) === true
+        && password === confirm_password
+    ) {
+        save_user();
+    } else {
+        next_step();
+        // alert("some error");
+    }
+}
+
+function password_validation(password) {
+    notifications_field.innerHTML = "";
+    // const password = "";
+    // console.log(password.match(/\s+/));
+    if (password === "") {
+        notifications_field.innerHTML = "Pass field is required!";
+        notifications_field.className = 'error_message'
+    } else if (password.length < 3) {
```

Comment:
В данном случае лучше юзвать свич кейс. Вообще вот когда и при каких условиях лучше использовать
Use switch instead of if when:

- You are comparing multiple possible conditions of an expression and the expression itself is non-trivial.
- You have multiple values that may require the same code.
- You have some values that will require essentially all of another value's execution, plus only a few statements.

Use if instead of switch when:

- You want to test for the truthiness of an expression.
- You only have a single affirmative test.
- You need to evaluate different expressions for each branch